monkey/common/agent_events/ping_scan_event.py

```diff
@@ -11,4 +15,4 @@ class PingScanEvent(AbstractAgentEvent):
         :param scan_data: The data collected from the ping scan
     """
 
-    scan_data: PingScanData
+    scan_data: Optional[PingScanData] = Field(default=None)
```

Comment:
Instead of this, can we just skip creating/sending `PingScanEvent`s if the data is None?

Reply:
This probably doesn't need to be Optional. I will check though